packages/extras/src/lib/components/Stars/Stars.svelte

```diff
@@ -0,0 +1,110 @@
+<script lang="ts">
+  import { AdditiveBlending, Color, ShaderMaterial, Spherical, Vector3 } from 'three'
+  import { T, forwardEventHandlers, useTask } from '@threlte/core'
+  import type { StarsEvents, StarsProps, StarsSlots } from './Stars.svelte'
+
+  import { fragmentShader } from './fragment'
+  import { vertexShader } from './vertex'
+
+  type $$Props = Required<StarsProps>
+  type $$Events = StarsEvents
+  type $$Slots = StarsSlots
+
+  export let count: $$Props['count'] = 5000
+  export let radius: $$Props['radius'] = 50
+  export let depth: $$Props['depth'] = 50
+  export let factor: $$Props['factor'] = 6
+  export let saturation: $$Props['saturation'] = 1.0
+	export let lightness: $$Props['lightness'] = 0.8
+  export let speed: $$Props['speed'] = 1
+  export let fade: $$Props['fade'] = true
+  export let opacity: $$Props['opacity'] = 1.0
+
+  const genStar = (r: number) => {
+    return new Vector3().setFromSpherical(
+      new Spherical(r, Math.acos(1 - Math.random() * 2), Math.random() * 2 * Math.PI)
+    )
+  }
```

Comment:
We can reduce garbage collection by reusing some three.js objects:
```suggestion
const vec3 = new Vector3()
const spherical = new Spherical()

  const genStar = (r: number) => {
    return vec3.setFromSpherical(
      spherical.set(r, Math.acos(1 - Math.random() * 2), Math.random() * 2 * Math.PI)
    )
  }
```